lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -4,42 +4,102 @@ module Floe
   class Workflow
     class Runner
       class Kubernetes < Floe::Workflow::Runner
-        attr_reader :namespace, :server, :token
+        TOKEN_FILE   = "/run/secrets/kubernetes.io/serviceaccount/token"
+        CA_CERT_FILE = "/run/secrets/kubernetes.io/serviceaccount/ca.crt"
 
         def initialize(options = {})
           require "awesome_spawn"
           require "securerandom"
           require "base64"
+          require "kubeclient"
           require "yaml"
 
-          @namespace = options.fetch("namespace", "default")
-          @server    = options.fetch("server", nil)
-          @token     = options.fetch("token", nil)
-          @token   ||= File.read(options["token_file"]) if options.key?("token_file")
+          kubeconfig_file    = ENV.fetch("KUBECONFIG", nil) || options.fetch("kubeconfig", File.join(Dir.home, ".kube", "config"))
+          kubeconfig_context = options.fetch("kubeconfig_context", nil)
+
+          kubeconfig  = Kubeclient::Config.read(kubeconfig_file) if File.exist?(kubeconfig_file)
+          kubecontext = kubeconfig.context(kubeconfig_context) if kubeconfig
+
+          token   = options.fetch("token", nil)
```

Comment:
Since `nil` is the default, this might be less wordy?  (Same with other `options` calls below).  `fetch` makes some sense for ENV (and I think rubocop requests it), but it's not really useful for Hashes

```suggestion
          token   = options["token"]
```

Reply:
:+1: rubocop complained about that on `ENV` and I went overboard the other way :laughing: